### PR TITLE
docs: add documentation for state-based migration

### PR DESCRIPTION
**Pull request by unknwon**
This PR adds the initial documentation for state-based migration.

Preview link: https://bytebase-com-git-joe-byt-1494-sbm-user-documentation-bytebase.vercel.app/docs/change-database/state-based-migration

### CODE REVIEW

File: content/docs/en/change-database/state-based-migration.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+title: State-based Migration
+---
+
+<hint-block type="warning">
+
+This feature is in beta and only available for projects that have enabled [GitOps workflow](../vcs-integration/enable-version-control-workflow).
+
+</hint-block>
+
+State-based migration is a declarative way to describe the desired state of the schema, it allows the schema management engine generate migration scripts automatically for any discrepancy found between the desired state and the actual schema.
```
Comment (tianzhou) [inline content/docs/en/change-database/state-based-migration.md:11]:
```suggestion
State-based migration is a declarative way to describe the desired state of the schema, it allows the schema management engine to generate migration scripts automatically for any discrepancy found between the desired state and the actual schema.
```

File: content/docs/en/change-database/state-based-migration.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+### Step 1 - Enable the setting
+
+Once enabled [GitOps workflow](../vcs-integration/enable-version-control-workflow), in the project's **Version Control > Schema change type** option, select **State-based**.
+
+![select-schema-change-type](/static/docs/en/change-database/state-based-migration/select-schema-change-type.webp)
+
+### Step 2 - Update schema file
+
+In our example with GitLab self-hosted as our VCS, we create a new schema file in the connected repository whose path is matching the **Schema path template** we have configured as `.{{DB_NAME}}__LATEST.sql`, where `{{DB_NAME}}` is `mydb`. It is noteable that we have also configured `bytebase` as the **Base directory**, so all files need to reside under this directory.
```
Comment (tianzhou) [inline content/docs/en/change-database/state-based-migration.md:29]:
```suggestion
In our example with GitLab self-hosted as our VCS, we create a new schema file in the connected repository whose path is matching the **Schema path template** we have configured as `.{{DB_NAME}}__LATEST.sql`, where `{{DB_NAME}}` is `mydb`. It is notable that we have also configured `bytebase` as the **Base directory**, so all files need to reside under this directory.
```

**Review by tianzhou — APPROVED** [review]
LGTM

**Comment by unknwon** [comment]
Merging as-is, happy to address any post-merge comments!

**Pull request merged**